replica/database.hh

```diff
@@ -1180,6 +1180,33 @@ public:
     friend class compaction_group;
 };
 
+// Smart table pointer that guards the table object
+// while it's being accessed asynchronously
+class table_ptr {
```

Comment:
nit, i'd suggest rename it to `table_holder` . and add a `hold()` helper to `table` class. the helper constructors calling into `find_column_family()` are nice but not quite necessary.

Reply:
renamed in v2